lib/Parser.js

```diff
@@ -1982,18 +1994,23 @@ class Parser extends Tapable {
 		);
 	}
 
-	getCommentOptions(range) {
+	parseCommentOptions(range) {
 		const comments = this.getComments(range);
 		if (comments.length === 0) return null;
 		const options = comments.map(comment => {
-			try {
-				let val = vm.runInNewContext(
-					`(function(){return {${comment.value}};})()`
-				);
-				return val;
-			} catch (e) {
-				return {};
+			const { value } = comment;
+			if (value && webpackCommentRegExp.test(value)) {
+				// try compile only if webpack options comment is present
+				try {
+					const val = vm.runInNewContext(`(function(){return {${value}};})()`);
+					return val;
+				} catch (e) {
+					throw Error(
```

Comment:
Don't throw an error here. Put the errors into a list an return `{ options: ..., errors: [...] }` from `parseCommentOptions`.

Elsewise this will change semantics and would be a breaking change. (because is would break the loop after the first error.

Example: `import(/* webpackMode broken */ /* webpackMode: "eager" */ "...")` (add this as test)